src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -94,9 +94,10 @@ public static partial class JsonSerializer
                             state.Current.JsonPropertyInfo.JsonPropertyName = propertyNameArray;
                         }
                     }
-
-                    state.Current.PropertyIndex++;
```

Comment:
Was this a bug? If so, how would that manifest (and do we have a test for it)?

Reply:
No it wasn't a bug. Now since we cache "misses" this LOC needed to move.